### PR TITLE
Prevent content duplication for initialEditorState function

### PR DESCRIPTION
**Pull request by fantactuka**
Content might be duplicated as it's assumed `initialEditorState` is only affects content once. The fix would be to check whether root is empty within `initialEditorState` but I think core should do that check instead

### CODE REVIEW

File: packages/lexical-rich-text/src/index.ts [hunk 1]
```diff
@@ -355,7 +354,12 @@ function initializeEditor(
         break;
       }
       case 'function': {
-        editor.update(initialEditorState, updateOptions);
+        editor.update(() => {
```
Comment (zurfyx) [inline packages/lexical-rich-text/src/index.ts:357]: If I'm reading this right this is inside an `editor.update` already

**Review by zurfyx — APPROVED** [review]
LGTM, I guess this can be a problem with double invoke effects?

File: packages/lexical-plain-text/src/index.ts [hunk 2]
```diff
@@ -146,7 +144,12 @@ function initializeEditor(
       }
 
       case 'function': {
-        editor.update(initialEditorState, updateOptions);
+        editor.update(() => {
+          const root = $getRoot();
+          if (root.isEmpty()) {
+            initialEditorState();
```
Comment (trueadm) [inline packages/lexical-plain-text/src/index.ts:150]: Can we also pass `editor` to this function so it's a bit more useful?

File: packages/lexical-rich-text/src/index.ts (hunk 1 above)
Comment (fantactuka) [reply packages/lexical-rich-text/src/index.ts:357]: Lol this made me sweating! I see what you mean here, but opened VS code and couldn't see it anymore :)
It's all Github with semi-collapsed lines, if you expand lines in between then `editor.update` only affects if branch, but not else part

File: packages/lexical-rich-text/src/index.ts [hunk 3]
```diff
@@ -330,8 +330,7 @@ function initializeEditor(
   } else if (initialEditorState === undefined) {
     editor.update(() => {
```
Comment (trueadm) [inline packages/lexical-rich-text/src/index.ts:331]: It still is in an update. This update.

Comment (fantactuka) [reply packages/lexical-rich-text/src/index.ts:331]: It ends before `else`, so `switch (typeof initialEditorState) {` part has its own .update call. (Need to expand collapsed lines)

Comment (trueadm) [reply packages/lexical-rich-text/src/index.ts:331]: Ah yeah, so confusing!

**Comment by trueadm** [comment]
Needs a prettier run, otherwise, awesome stuff!

**Comment by trueadm** [comment]
I think you might have missed updating the Flow definition files?

**Comment by trueadm** [comment]
It just occurred to me that this PR might not cause the root element to have the latest content. If the root is not empty, we should probably be doing `setEditorState(editorState)` to force a full reconciliation, in-case the DOM node for the content editable is empty (because React cleared it).

**Pull request merged**